### ISSUE TITLE
Logger revisions

### ISSUE THREAD

**Issue by nsheff** [epigen/looper#40]
I'm trying to use the current version (0.4-rc2 +changes). Logger needs cleanup! We need to reign it in just a bit at least for default purposes.

e.g., this is what I'm getting now, which is going to be incomprehensible to a typical user:
```
WARNING:root:None as logs destination appears to be neither a filepath nor a stream.
2017-02-16 11:43:08,578 looper.models.Project models : 265 - [INFO] > Instantiating Project using config file /home/ns5bc/code//mitodyn/metadata/config.yaml
INFO:looper.models.Project:Instantiating Project using config file /home/ns5bc/code//mitodyn/metadata/config.yaml
2017-02-16 11:43:08,579 looper.models.Project models : 454 - [INFO] > Loading LOOPERENV: /home/ns5bc/.local/lib/python2.7/site-packages/looper/submit_templates/default_looperenv.yaml
INFO:looper.models.Project:Loading LOOPERENV: /home/ns5bc/.local/lib/python2.7/site-packages/looper/submit_templates/default_looperenv.yaml
2017-02-16 11:43:08,590 looper.models.Project models : 289 - [INFO] > Updating compute settings (looper environment) based on file '/home/ns5bc/code/looperenv/rivanna.yaml'
INFO:looper.models.Project:Updating compute settings (looper environment) based on file '/home/ns5bc/code/looperenv/rivanna.yaml'
2017-02-16 11:43:08,591 looper.models.Project models : 454 - [INFO] > Loading LOOPERENV: /home/ns5bc/code/looperenv/rivanna.yaml
INFO:looper.models.Project:Loading LOOPERENV: /home/ns5bc/code/looperenv/rivanna.yaml
2017-02-16 11:43:08,601 looper.models.Project models : 478 - [ERROR] > Can't load looperenv config file '/home/ns5bc/code/looperenv/rivanna.yaml'
ERROR:looper.models.Project:Can't load looperenv config file '/home/ns5bc/code/looperenv/rivanna.yaml'
2017-02-16 11:43:08,601 looper.models.Project models : 479 - [ERROR] > TypeErroradd_entries() takes exactly 2 arguments (1 given)
ERROR:looper.models.Project:TypeErroradd_entries() takes exactly 2 arguments (1 given)
2017-02-16 11:43:08,601 looper.models.Project models : 294 - [INFO] > Establishing project compute settings
INFO:looper.models.Project:Establishing project compute settings
2017-02-16 11:43:08,602 looper.models.Project models : 515 - [INFO] > Loading compute settings default
INFO:looper.models.Project:Loading compute settings default
```

The logger messages are a bit too verbose for my tastes. I propose:

1. lines are way too long. Default formatter should be simple with none (or fewer) of the appendages (like level, lcoation, etc; just the message). Only by request should we add the extra information.
2. messages are printed twice?
3. `compute` section is now spitting out a bunch of logger-related attributes? eg: `{'_force_nulls': False, '_logger': <logging.Logger object at 0x3b60b50>,`
4. why can looper no longer read the `looperenv` file?
```
2017-02-16 11:56:16,843 looper.models.Project models : 478 - [ERROR] > Can't load looperenv config file '/home/ns5bc/code/looperenv/rivanna.yaml'
ERROR:looper.models.Project:Can't load looperenv config file '/home/ns5bc/code/looperenv/rivanna.yaml'
```
5. magic numbers need explanation, eg. what is ",820" in this? `11:56:16,820` and what is 265 in this? `ls.Project models : 265 `
6. Why does processing not stop at the error (843 or 479)?

And finally, this version is now broken and will not run on this project (0.4 worked).
```
Traceback (most recent call last):
  File "/home/ns5bc/.local/bin/looper", line 9, in <module>
    load_entry_point('looper==0.4', 'console_scripts', 'looper')()
  File "/home/ns5bc/.local/lib/python2.7/site-packages/looper/looper.py", line 674, in main
    looperenv_file=getattr(args, 'env', None))
  File "/home/ns5bc/.local/lib/python2.7/site-packages/looper/models.py", line 310, in __init__
    self.parse_config_file(subproject)
  File "/home/ns5bc/.local/lib/python2.7/site-packages/looper/models.py", line 359, in parse_config_file
    self.add_entries(self.config)
  File "/home/ns5bc/.local/lib/python2.7/site-packages/looper/models.py", line 142, in add_entries
    self[key] = value
  File "/home/ns5bc/.local/lib/python2.7/site-packages/looper/models.py", line 174, in __setitem__
    self.__dict__[key].add_entries()
TypeError: add_entries() takes exactly 2 arguments (1 given)
```

**Comment by nsheff** [comment]
Just realized I'm on the vreurter-0.5 branch (not the 0.4-rc2 + changes as I originally stated).

**Comment by nsheff** [comment]
I have encountered another error in the current dev version:
`
WARNING:looper.models.Sample:Config lacks entry for data_source key: 'mitodyn' (in column: 'data_source')
Traceback (most recent call last):
  File "/home/ns5bc/.local/bin/looper", line 9, in <module>
    load_entry_point('looper==0.4', 'console_scripts', 'looper')()
  File "/home/ns5bc/.local/lib/python2.7/site-packages/looper/looper.py", line 677, in main
    _LOGGER.info("Results subdir: " + prj.metadata.results_subdir)
AttributeError: 'NoneType' object has no attribute 'info'
`

**Comment by nsheff** [comment]
The current version in master is also not correctly processing derived columns; so something went wonky in the updates. 0.4-rc2 works, but current master branch is broken. the derived column is stuck on data_source for some reason...

**Comment by vreuter** [comment]
1. Agreed. I'll omit logger name, which will substantially shorten most of those. Knowing line number and module is much shorter, and the benefit of class name is really a subset of that combination. I can also eliminate the spaces around line number. How do you feel about lowercase rather than uppercase for level? I think level is still helpful and makes it stand out, but I guess it could intimidate users?
2. Which messages are printed twice? The duplicate line number that I see there is from line 454, which originates from two different calls, giving two different filenames.
3. Only one of those is logger-related; the other(s) are the behavioral-related parameters on AttributeDict that then pop up in all of the looper-specific data types that inherit from AttributeDict. This could be changed by tweaking the implementation of __str__ and/or __repr__ on AttributeDictr.
4. I don't have an immediate answer; I'll have to dig more on that.
5. Those aren't magic numbers. 820 there is milliseconds on logger message time; that should be eliminated and I will do so. 265 is the line within the models module from which the message originated. As mentioned in pt.1, I'd like to have that, but I'm happy to eliminate it for the sake of logging message brevity and clarity from the user side.
6. Processing doesn't stop at many of the errors because there are several places in this code that have very generic Exception types in their catch/except clause with no re-raising. I don't like that and agree that processing should stop, but I left those as-was in many places because I thought perhaps that was intended behavior?

**Comment by nsheff** [comment]
> Which messages are printed twice? 
All messages are in there twice; ie, "Instantiating Project using config file" in this:
`2017-02-16 11:43:08,578 looper.models.Project models : 265 - [INFO] > Instantiating Project using config file /home/ns5bc/code//mitodyn/metadata/config.yaml
INFO:looper.models.Project:Instantiating Project using config file /home/ns5bc/code//mitodyn/metadata/config.yaml`

**Comment by nsheff** [comment]
> Agreed. I'll omit logger name, which will substantially shorten most of those. Knowing line number and module is much shorter, and the benefit of class name is really a subset of that combination. I can also eliminate the spaces around line number. How do you feel about lowercase rather than uppercase for level? I think level is still helpful and makes it stand out, but I guess it could intimidate users?

I agree that level is still helpful and uppercase is fine. and it's short. message should move to front? maybe format like this:

[INFO] Message... (module:linenum)

> Those aren't magic numbers.
Ok yeah milliseconds is overkill :). I think all the time can probably be removed by default. looper usually executes in real time. this will be useful in the future for other projects that take hours to complete, though.

> Processing doesn't stop at many of the errors because there are several places in this code that have very generic Exception types in their catch/except clause with no re-raising. I don't like that and agree that processing should stop, but I left those as-was in many places because I thought perhaps that was intended behavior?

Ok perfect, this is something we can clean up over time.

**Comment by vreuter** [comment]
As discussed earlier with @nsheff , verbosity control should become more intuitive. If a user specifies one of the `logging` builtin level names on the command line, then that should be used as-is. Otherwise, though, the more general use case should be specification via small integral values. Furthermore, the relationship between value given and logging verbosity should be one of positive correlation, not negative as the bultin `logging` module does it. From the user side, that accords much more with intuition.

**Comment by vreuter** [comment]
Track logging updates here: https://github.com/epigen/looper/issues/49

**Comment by nsheff** [comment]
And moved errors stopping question to issue #51

**Comment by nsheff** [comment]
I believe everything in this issue is either addressed in v0.5, or has been moved to a separate issue, so I'm closing this one.
